File: daa/src/main/java/edu/sjsu/cs/cs255/lists/LinkedList.java
package edu.sjsu.cs.cs255.lists;


class Node<T> {
    T data;
    Node<T> next;

    Node(T data) {
        this.data = data;
        this.next = null;
    }
}

public class LinkedList<T> {

    Node root = null;

    void insert(T data) {
        Node<T> newNode = new Node<>(data);
        if (root == null) {
            root = newNode;
            return;
        }
        Node ptr = root;
        while (ptr.next != null) {
            ptr = ptr.next;
        }
        ptr.next = newNode;
    }

    void traverse() {
        Node ptr = root;
        while (ptr != null) {
            System.out.print(ptr.data + " ");
            ptr = ptr.next;
        }
        System.out.println();
    }
}

File: cpp-lang/data-structures/LinkedList.cpp
//
// Created by Sushant Mane on 10/24/19.
//

#include <iostream>
#include "LinkedList.h"

void LinkedList::addNode(int data) {
    Node *n = new Node();
    n->data = data;
    n->next = nullptr;
    if (!head) {
        head = n;
        return;
    }
    Node *ptr = head;
    while (ptr->next) {
        ptr = ptr->next;
    }
    ptr->next = n;
}

void LinkedList::print() {
    Node *ptr = head;
    while (ptr) {
        std::cout << ptr->data << " ";
        ptr = ptr->next;
    }
}
